### PR TITLE
feat: address invalid schnorr signature error

### LINKED ISSUE #690: [Feature]: Address invalid Schnorr signature error
# Feature - Address invalid Schnorr signature error

## 1. Description

During the pre-demo, we sometimes got an error from bitcoin-core rejecting the transaction with the error message `"non-mandatory-script-verify-flag (Invalid Schnorr signature)"`. This happens randomly during our tests, but it appears to depend on the parity of the aggregate key.

### 1.1 Context & Purpose

Well, we cannot be a bridge if we cannot always sign for the transactions that we are supposed to sign for.

## 2. Technical Details:

Figure out the cause of the issue and solve it. Find a way.

### 2.1 Acceptance Criteria:
   - [ ] The signers never get that error from bitcoin core for their transactions.

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes #690 

## Changes

* Update the version of WSTS that we use and make the required updates given the interface changes. It points to the HEAD of https://github.com/Trust-Machines/wsts/pull/94.
* Fix a conversion bug when going from `BurnchainHeaderHash` to our internal type.
* Fix a bug where we did not create the state machine correctly when signing bitcoin transactions.

## Testing Information

There adds an end-to-end integration test where "signers" run DKG and successfully sign and submit a bitcoin transaction. This test should pass if you comment out the lines in the tx-coordinator and tx-signer that have
```rust
self.context
    .signal(TxCoordinatorEvent::MessageGenerated(msg).into())?;
```

Maybe I'll update it to also test the submission of the stacks transaction, which it currently ignores.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: signer/tests/integration/transaction_coordinator.rs [hunk 1]
```diff
@@ -607,3 +635,310 @@ async fn run_dkg_from_scratch() {
...
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    faucet.generate_blocks(1);
+
+    // 5. Once they are all running, signal that DKG should be run. We
+    //    signal them all because we do not know which one is the
+    //    coordinator.
+
+    tokio::time::sleep(Duration::from_secs(3)).await;
```
Comment (cylewitruk) [inline signer/tests/integration/transaction_coordinator.rs:863]: Instead of a fixed 3s you could wait for `BitcoinBlockObserved` from all the signers.

**Review by cylewitruk — COMMENTED** [review]
Looks good in general, just a few typos/comments/suggestions

File: signer/src/transaction_coordinator.rs [hunk 2]
```diff
@@ -274,13 +276,14 @@ where
...
     /// fulfilling pending deposit and withdraw requests.
-    #[tracing::instrument(skip(self, signer_public_keys, aggregate_key))]
+    #[tracing::instrument(skip(self, aggregate_key, signer_public_keys))]
     async fn construct_and_sign_bitcoin_sbtc_transactions(
         &mut self,
         bitcoin_chain_tip: &model::BitcoinBlockHash,
         aggregate_key: &PublicKey,
         signer_public_keys: &BTreeSet<PublicKey>,
     ) -> Result<(), Error> {
+        tracing::info!("Fetching the stacks chain tip");
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/src/transaction_coordinator.rs]: https://github.com/stacks-network/sbtc/pull/756/commits/7a8bb519f5382c1a57f212520db6d1d5ae79ff94

File: signer/src/transaction_coordinator.rs [hunk 3]
```diff
@@ -274,13 +276,14 @@ where
 
     /// Construct and coordinate WSTS signing rounds for sBTC transactions on Bitcoin,
     /// fulfilling pending deposit and withdraw requests.
-    #[tracing::instrument(skip(self, signer_public_keys, aggregate_key))]
+    #[tracing::instrument(skip(self, aggregate_key, signer_public_keys))]
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/src/transaction_coordinator.rs]: https://github.com/stacks-network/sbtc/pull/756/commits/7a8bb519f5382c1a57f212520db6d1d5ae79ff94

File: signer/src/testing/transaction_signer.rs [hunk 4]
```diff
@@ -691,19 +691,18 @@ where
...
-            .run_signing_round(bitcoin_chain_tip, txid, &msg)
+            .run_signing_round(bitcoin_chain_tip, txid, &msg, SignatureType::Schnorr)
             .await;
 
         // Let's check the signature using the secp256k1 types.
         let sig = secp256k1::schnorr::Signature::from_slice(&signature.to_bytes()).unwrap();
         let msg_digest = secp256k1::Message::from_digest(msg);
-        let pk = aggregate_key.signers_tweaked_pubkey().unwrap();
-        let x_only_pk = secp256k1::XOnlyPublicKey::from(&pk);
+        // let pk = aggregate_key;
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/src/testing/transaction_signer.rs]: https://github.com/stacks-network/sbtc/pull/756/commits/7a8bb519f5382c1a57f212520db6d1d5ae79ff94

**Review by cylewitruk — APPROVED** [review]
Looks good, the suggestions are ofc optional :unicorn:

File: signer/tests/integration/transaction_coordinator.rs [hunk 5]
```diff
@@ -607,3 +635,310 @@ async fn run_dkg_from_scratch() {
...
+        deposit_script: deposit_request.deposit_script.to_hex_string(),
+        reclaim_script: deposit_request.reclaim_script.to_hex_string(),
+    };
+    let _ = deposit_api::create_deposit(emily_client.config(), body)
+        .await
+        .unwrap();
+
+    faucet.generate_blocks(1);
+
+    tokio::time::sleep(Duration::from_secs(5)).await;
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/tests/integration/transaction_coordinator.rs:1304]: I think I'm going to come back to this when we finally enable this test.

File: signer/tests/integration/transaction_coordinator.rs (hunk 1 above)
Comment (djordon) [reply signer/tests/integration/transaction_coordinator.rs:1261]: Same for the above, let's come back to this when we enable this test.

**Pull request merged**